test/native/perfEventsLinuxTest.cpp

```diff
@@ -0,0 +1,269 @@
+}
+
+TEST_CASE(ForName_Breakpoint_Addr_DecOffset) {
+  PerfEventType* event_type = PerfEventType::forName("mem:0x123+22000");
+  ASSERT_BP(event_type, HW_BREAKPOINT_RW, 22291, 1, 0);
+}
+
+TEST_CASE(ForName_Breakpoint_Addr_HexOffset) {
+  PerfEventType* event_type = PerfEventType::forName("mem:0x123+0x22000");
+  ASSERT_BP(event_type, HW_BREAKPOINT_RW, 0x22123, 1, 0);
+}
+
+TEST_CASE(ForName_Breakpoint_Addr_HexOffset_DecLen) {
+  PerfEventType* event_type = PerfEventType::forName("mem:0x123+0x22000/16");
+  ASSERT_BP(event_type, HW_BREAKPOINT_RW, 0x22123, 16, 0);
+}
+
+TEST_CASE(ForName_Breakpoint_Addr_HexOffset_HexLen) {
+  PerfEventType* event_type = PerfEventType::forName("mem:0x123+0x22000/0x16");
+  ASSERT_BP(event_type, HW_BREAKPOINT_RW, 0x22123, 0x16, 0);
+}
+
+TEST_CASE(ForName_Breakpoint_Addr_HexOffset_HexLen_R) {
+  PerfEventType* event_type = PerfEventType::forName("mem:0x123+0x22000/0x16:r");
+  ASSERT_BP(event_type, HW_BREAKPOINT_R, 0x22123, 0x16, 0);
+}
+
+TEST_CASE(ForName_Breakpoint_Addr_HexOffset_HexLen_W) {
+  PerfEventType* event_type = PerfEventType::forName("mem:0x123+0x22000/0x16:w");
+  ASSERT_BP(event_type, HW_BREAKPOINT_W, 0x22123, 0x16, 0);
+}
+
+TEST_CASE(ForName_Breakpoint_Addr_HexOffset_HexLen_X) {
+  PerfEventType* event_type = PerfEventType::forName("mem:0x123+0x22000/0x16:x");
+  ASSERT_BP(event_type, HW_BREAKPOINT_X, 0x22123, 0x16, 0);
+}
+
+TEST_CASE(ForName_Breakpoint_Addr_HexOffset_HexLen_InvalidDefaultRW) {
+  PerfEventType* event_type = PerfEventType::forName("mem:0x123+0x22000/0x16:i");
+  ASSERT_BP(event_type, HW_BREAKPOINT_RW, 0x22123, 0x16, 0);
+}
+
+TEST_CASE(ForName_Breakpoint_Addr_HexOffset_HexLen_X_Arg) {
+  PerfEventType* event_type = PerfEventType::forName("mem:0x123+0x22000/0x16:x{3}");
+  ASSERT_BP(event_type, HW_BREAKPOINT_X, 0x22123, 0x16, 3);
+}
+
+TEST_CASE(ForName_Breakpoint_Addr_HexOffset_HexLen_RW) {
+  PerfEventType* event_type = PerfEventType::forName("mem:0x123+0x22000/0x16:rw");
+  ASSERT_BP(event_type, HW_BREAKPOINT_RW, 0x22123, 0x16, 0);
+}
+
+TEST_CASE(ForName_Breakpoint_Addr_HexOffset_HexLen_RX) {
+  PerfEventType* event_type = PerfEventType::forName("mem:0x123+0x22000/0x16:rx");
+  ASSERT_BP(event_type, HW_BREAKPOINT_RW, 0x22123, 0x16, 0);
+}
+
+TEST_CASE(ForName_Breakpoint_Addr_HexOffset_HexLen_WX) {
+  PerfEventType* event_type = PerfEventType::forName("mem:0x123+0x22000/0x16:wx");
+  ASSERT_BP(event_type, HW_BREAKPOINT_RW, 0x22123, 0x16, 0);
+}
+
+TEST_CASE(ForName_Breakpoint_Addr_HexOffset_HexLen_X_Known) {
+  const __u64 mmap_addr = (__u64)(uintptr_t)dlsym(RTLD_DEFAULT, "mmap");
+  PerfEventType* event_type = PerfEventType::forName("mem:mmap+0x22000/0x16:x");
+
+  ASSERT_NE(mmap_addr, 0);
+  ASSERT_BP(event_type, HW_BREAKPOINT_X, mmap_addr + 0x22000, 0x16, 2);
+}
+
+TEST_CASE(ForName_Breakpoint_Symbol) {
+  PerfEventType* event_type = PerfEventType::forName("mem:strcmp");
+  const __u64 strcmp_addr = (__u64)(uintptr_t)dlsym(RTLD_DEFAULT, "strcmp");
```

Comment:
Just curious: is it intentional that the order of forName/dlsym differs in the above tests?

Reply:
It is not, changed to be consistent.